ui/app/components/app/account-menu/tests/account-menu.test.js

```diff
@@ -0,0 +1,208 @@
+import React from 'react'
+import assert from 'assert'
+import sinon from 'sinon'
+import configureMockStore from 'redux-mock-store'
+import { mountWithRouter } from '../../../../../../test/lib/render-helpers'
+import AccountMenu from '../index'
+import { Provider } from 'react-redux'
+
+describe('Account Menu', async () => {
+
+  let wrapper
+
+  const mockStore = {
+    metamask: {
+      provider: {
+        type: 'test',
+      },
+      preferences: {
+        useNativeCurrencyAsPrimaryCurrency: true,
+      },
+    },
+  }
+
+  const store = configureMockStore()(mockStore)
+
+  const props = {
+    isAccountMenuOpen: true,
+    addressConnectedDomainMap: {},
+    accounts: [
+      {
+        address: '0xAddress',
+        name: 'Account 1',
+        balance: '0x0',
+      },
+      {
+        address: '0xImportedAddress',
+        name: 'Imported Account 1',
+        balance: '0x0',
+      },
+    ],
+    keyrings: [
+      {
+        type: 'HD Key Tree',
+        accounts: [
+          '0xAdress',
+        ],
+      },
+      {
+        type: 'Simple Key Pair',
+        accounts: [
+          '0xImportedAddress',
+        ],
+      },
+    ],
+    prevIsAccountMenuOpen: false,
+    lockMetamask: sinon.spy(),
+    showAccountDetail: sinon.spy(),
+    showRemoveAccountConfirmationModal: sinon.spy(),
+    toggleAccountMenu: sinon.spy(),
+    history: {
+      push: sinon.spy(),
+    },
+
+  }
+
+  before(() => {
+    wrapper = mountWithRouter(
+      <Provider store={store}>
+        <AccountMenu.WrappedComponent {...props} />
+      </Provider>, store
+    )
+  })
+
+  afterEach(() => {
+    props.toggleAccountMenu.resetHistory()
```

Comment:
For my own reference: 
> Since sinon@5.0.0, the sinon object is a default sandbox. Unless you have a very advanced setup or need a special configuration, you probably want to only use that one.

Will change the behaviors of the `resetHistory/restore/etc` to get a more streamlied approach.